Hestia/app/src/main/java/hestia/UI/AddDeviceInfo.java

```diff
@@ -79,14 +85,17 @@ private EditText createEditText(String key, LinearLayout.LayoutParams params, in
     public void onClick(View view) {
         switch (view.getId()) {
             case R.id.confirm_button:
-                HashMap<String, String> h = getFieldValues();
-                if(h==null) {
+                JsonObject requiredInfo = this.getRequiredInfo();
+                if(requiredInfo==null) {
                     Toast.makeText(getContext(), "One or more empty values were entered."
                             , Toast.LENGTH_SHORT).show();
                     break;
                 }
-                new PostDeviceTask(h).execute();
-                Toast.makeText(content, h.toString(), Toast.LENGTH_SHORT).show();
+                String path = BackendInteractor.getInstance().getPath() + "devices/";
```

Comment:
We need to change all of these path references, but this is not an issue of the current PR, I just wanted to mention it here as well.